rust/private/rustc.bzl

```
@@ -1149,6 +1149,31 @@ def construct_arguments(
 
     return args, env
 
+def _get_dynamic_libraries_for_runtime(cc_linking_context, linking_statically):
+    libraries = []
+    if not cc_linking_context:
+        return libraries
+
+    for linker_input in cc_linking_context.linker_inputs.to_list():
+        libraries.extend(linker_input.libraries)
+
+    dynamic_libraries_for_runtime = []
+    for library in libraries:
+        artifact = _get_dynamic_library_for_runtime_or_none(library, linking_statically)
+        if artifact != None:
+            dynamic_libraries_for_runtime.append(artifact)
+
+    return dynamic_libraries_for_runtime
+
+def _get_dynamic_library_for_runtime_or_none(library, linking_statically):
```

nit: this and/or add docstrings to make the types clearer 😅 
```suggestion
def _get_dynamic_library_for_runtime_or_none(library, is_linking_statically):
```